addons/web_editor/static/src/js/wysiwyg/widgets/chatgpt_alternatives_dialog.js

```diff
@@ -39,7 +39,9 @@ export class ChatGPTAlternativesDialog extends ChatGPTDialog {
             }],
             messages: [],
             alternativesMode: '',
+            currentBatchId: null,
```

Comment:
The commit message gives me the impression that there was already some sort of batch handling and aborting going on, and that you found one particular use case in which the existing system didn't work. Under that assumption, I'm surprised to see an entire batch handling system popping up in the diff. The most likely explanation from my point of view is probably that there was no aborting going on in the first place and that this commit introduces a mechanism to do it ? If that's the case then I suggest rewriting the commit message.

Reply:
Indeed there were batches (3 alternatives were always generated at a time), but there was no way to abort so you could end up with as many as 6 alternatives instead of 3 if you launched a new batch before the end of the first. I'll rephrase.

```
This implements a way to abort the generation of a batch of alternatives
by the AI Copywriter. This is needed because if a batch was being
generated when the user clicked one of the buttons to select a mode, it
continued to be generated and to update the state while the new batch
was in the making, resulting in a potential total of more alternatives
than requested.
```